event-sauce/tests/action_builder.rs

```diff
@@ -25,32 +25,46 @@ impl AggregateAction<UserEventData> for User {
         if let Some(ref data) = event.data {
             match data {
                 UserEventData::UserCreated(_) => {
-                    // let create_event = event.clone().into_event::<UserCreated>(Some(data.clone()));
-                    let create_event = event
-                        .clone()
-                        .try_into_variant::<UserCreated>()
-                        // TODO: Better error variant
-                        .map_err(|_e| EventError::Infallible())?;
+                    let create_event =
+                        event
+                            .clone()
+                            .try_into_variant::<UserCreated>()
+                            .map_err(|_| {
+                                EventError::ConversionError(
+                                    "Event<UserEventData>",
+                                    "Event<UserCreated>",
```

Comment:
Not something to change for this PR, but for future code you could use [`type_name`](https://doc.rust-lang.org/std/any/fn.type_name.html) in these cases.

Reply:
Ah, I was looking for something like this - thank!